src/ansys/simai/core/data/model_configuration.py

```diff
@@ -0,0 +1,257 @@
+# Copyright (C) 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from dataclasses import asdict, dataclass
+from typing import TYPE_CHECKING, Any, Literal, Optional
+
+if TYPE_CHECKING:
+    from ansys.simai.core.data.projects import Project
+
+
+@dataclass
+class GlobalCoefficientUnit:
+    """Single Global Coefficient field.
+
+    Args:
+        formula: the Global Coefficient formula.
+        name: the name of the Global Coefficient.
+    """
+
+    formula: str
+    name: str
+
+
+@dataclass
+class ModelInput:
+    """The inputs of a model.
+
+    Args:
+        surface: the input surface variables.
+        boundary_conditions: the boundary conditions.
+    """
+
+    surface: list[str] = None
+    boundary_conditions: list[str] = None
+
+
+@dataclass
+class ModelOutput:
+    """The outputs of a model.
+
+    Args:
+        surface: the output surface variables.
+        volume: the output volume variables.
+    """
+
+    surface: list[str] = None
+    volume: list[str] = None
+
+
+@dataclass
+class ModelConfiguration:
+    """Configures the build of a model.
+
+    Args:
+        project: the project of the configuration.
+        build_preset: indicates the build duration. Available options:
+
+                    | *debug*: < 30 min
+
+                    | *short*: < 24 hours
+
+                    | *default*: < 2 days, default value.
+
+                    | *long*: < 1 week
+        continuous: indicates if continuous learning is enabled. Default is False.
+        input: the inputs of the model.
+        output: the outputs of the model.
+        global_coefficients: the Global Coefficients of the model.
+
+    Example:
+        Define a new configuration and launch a build.
+
+        .. code-block:: python
```

Comment:
That's a pretty big example, it's gonna get outdated and broken fast...
In fact it's already broken (`simulation_volume` is not defined).

Reply:
I merged with `doa` so that the example is up-to-date. True that is big, however, moving into examples could be done in a later stage.